### DISCUSSION TITLE
Using .plx files sorted and unsorted for CellExplorer

### DISCUSSION THREAD

**Discussion by ajcheng1** [petersenpeter/CellExplorer#115] [General]
I did see #79 but I was wondering if there is a way to preserve spike waveforms when using .plx data since I need the single-cell analysis . The .plx data  can be in its unsorted raw form and then using offline sorter, it could sort the spike into another .plx data. However, CellExplorer can't take .plx formats.
I am currently trying to convert .plx files to .nwb but the python programs I have been using/editing have not been the most cooperative. Is there an alternative way I can go about to extract certain characteristics (like neuron type, excitatory vs inhibitory) from my data?

**Comment by petersenpeter** [comment]
Hi Alvin

Please see the script called [loadSpikes](https://github.com/petersenpeter/CellExplorer/blob/main/calc_CellMetrics/loadSpikes.m) in the CellExplorer repo and the documentation of the [spikes struct](https://cellexplorer.org/datastructure/data-structure-and-format/#spikes). You should be able to edit the script to make it compatible with plx-sorted data. If there is a plx package for Matlab that can read the spike-sorted data, CellExplorer should be able to do it. 

That said, it works best with the raw data as a binary file (or in a similar format).
